rholang/src/main/scala/coop/rchain/rholang/interpreter/matcher/package.scala

```diff
@@ -25,6 +25,11 @@ package object matcher {
 
   type MatcherMonadT[F[_], A] = StateT[StreamT[F, ?], FreeMap, A]
 
+  import coop.rchain.rholang.interpreter.matcher.StreamT
+
+  implicit def matcherMonadCostLog[F[_]: Monad: _cost](): _cost[MatcherMonadT[F, ?]] =
+    ntCostLog(λ[F ~> MatcherMonadT[F, ?]](fa => StateT.liftF(StreamT.liftF(fa))))
```

Comment:
This is required to derive a `_cost` instance for the `MatcherMonad` used by the `SpatialMatcher`. The reason could be the `StreamT`, or better said, the lack of some required derivation rule. I guess this could be investigated and improved in a separate PR